### PR TITLE
Handle cancel button clicks in forms

### LINKED ISSUE #114: Handle Cancel button in forms
- The Cancel button in ActionButtons is currently a stub

Screen Shot 2021-02-09 at 8 30 57 PM

- Make sure the user wants to abandon their progress ([Figma](https://www.figma.com/file/1C71g93ZA1yFWchMa4id7J/MU-Map-2020-21?node-id=712%3A2267)):
Screen Shot 2021-02-09 at 8 30 21 PM

### PR DESCRIPTION
**Pull request by tedspare**
# Summary

- Propagate cancel event from forms up to App
- Add an "unsaved changes" warning component which keeps the form open if the user clicks "Stay"

I'm not sure if my approach is the best, especially in `toggleSidebar()` which is getting a bit messy. Any suggestions?

## Test Plan

Run in browser

## Related Issues

Which issues does this PR resolve/work on?

Closes #114

### CODE REVIEW

File: frontend/src/App/App.tsx [hunk 1]
```diff
@@ -22,7 +23,8 @@ function App() {
   const [murals, setMurals] = useState<any>([]);
   const [searchResult, setSearchResult] = useState<any>([]);
   const [signInError, setSignInError] = useState<string>("");
-  const [activeForm, setActiveForm] = useState<string>("mural");
+  const [activeForm, setActiveForm] = useState<string>("");
```
Comment (JDziewonski98) [inline frontend/src/App/App.tsx:26]: I'm thinking this should be of an Enum type {Mural, Collection, ...} ?

**Review by JDziewonski98 — APPROVED** [review]
LGTM, I agree that there is probably a better design to toggleSidebar, but I cannot think of one... Only suggestion is to consider using an Enum as per my comment, and maybe make toggleSidebar take Enum || null

**Comment by tedspare** [comment]
> LGTM, I agree that there is probably a better design to toggleSidebar, but I cannot think of one... Only suggestion is to consider using an Enum as per my comment, and maybe make toggleSidebar take Enum || null

this is definitely an improvement

another option (which would take a major refactor) is to wrap the forms in a common `Form` component which provides save and cancel functionality

now that I'm writing this, I'm realizing that functionality would fit inside of `ActionButtons` quite nicely

**Review by zhenglinxiao — APPROVED** [review]
Looks great!

**Pull request merged**